dashboard/app/models/pd/teacher_application.rb

```diff
@@ -133,6 +134,31 @@ def application_hash
     application ? JSON.parse(application) : {}
   end
 
+  def accepted_workshop=(workshop_name)
+    if workshop_name.nil?
+      accepted_program.try(:destroy)
+      return
+    end
+
+    params = {
+      workshop_name: workshop_name,
+      course: selected_course,
+      user_id: user_id,
+      teacher_application_id: id
+    }
+
+    if accepted_program
+      accepted_program.update! params
+    else
+      self.accepted_program = Pd::AcceptedProgram.find_or_create_by! params
+    end
+  end
+
+  def accepted_workshop
```

Comment:
In addition to the one we're already tracking (https://trello.com/c/AAtZWL1c/39-create-registrations-for-ecs-csd-teachers-based-on-sarah-s-list)? If so, can you be more specific as to what you want to pull out and track?